database/sql/headers.go

```diff
@@ -480,3 +483,46 @@ func (h *HeadersDb) getMerkleRootConfirmation(item domains.MerkleRootConfirmatio
 
 	return confirmation, nil
 }
+
+// GetMerkleRoots method will retrieve as many merkleroots as batchSize from the db from lastEvaluatedKey exclusive
+func (h *HeadersDb) GetMerkleRoots(batchSize int, lastEvaluatedKey string) ([]*dto.DbMerkleRoot, error) {
+	var merkleroots []*dto.DbMerkleRoot
+	lastEvaluatedHeight, err := h.getLastEvaluatedMerklerootHeight(lastEvaluatedKey)
+	if err != nil {
+		return nil, err
+	}
+
+	err = h.db.Select(&merkleroots, h.db.Rebind(sqlMerkleRootsFromHeight), &lastEvaluatedHeight, batchSize)
+	if err != nil && !errors.Is(err, sql.ErrNoRows) {
+		return nil, err
+	}
+
+	return merkleroots, nil
+}
+
+func (h *HeadersDb) getLastEvaluatedMerklerootHeight(lastEvaluatedKey string) (*int32, error) {
```

Comment:
Why don't return "just" a normal `int32` (not pointer)?
- The return `-1, nil` for the first case would be more straightforward 
- Others would be more confused
- For the cases when error occurs you can always return `0, err`

Reply:
agree